# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/security/test/general/test_same_site_cookies_laxByDefault.html
# Commit: d01a1a055fec
# Full Hash: d01a1a055fece12b906b938954f984bd69278ef7
# Author: Andrea Marchesini <amarchesini@mozilla.com>
# Date: 2025-04-30 20:49:53
# Regressor Bug: 1960824
# File Overlap Count: 1
# Description:
#   Bug 1960824 - Introduce nsICookie::SAMESITE_UNSET, r=edgul,webdriver-reviewers,extension-reviewers,cookie-reviewers,valentin,Sasha,robwu
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D246638
# ==============================================================================

diff -r 307ff391dd02 -r d01a1a055fec dom/security/test/general/test_same_site_cookies_laxByDefault.html
--- a/dom/security/test/general/test_same_site_cookies_laxByDefault.html	Wed Apr 30 08:53:12 2025 +0000
+++ b/dom/security/test/general/test_same_site_cookies_laxByDefault.html	Wed Apr 30 08:59:18 2025 +0000
@@ -29,19 +29,29 @@
     const {sendAsyncMessage} = this;
     let cookies = { test: null, test2: null, test3: null };
 
+    function sameSiteToString(sameSite) {
+      switch (sameSite) {
+        case Ci.nsICookie.SAMESITE_NONE: return "none";
+        case Ci.nsICookie.SAMESITE_LAX: return "lax";
+        case Ci.nsICookie.SAMESITE_STRICT: return "strict";
+        case Ci.nsICookie.SAMESITE_UNSET: return "unset";
+        default: return "ERROR!";
+      }
+    }
+
     for (let cookie of Services.cookies.cookies) {
       if (cookie.host != "example.com") continue;
 
       if (cookie.name == "test" && cookie.value == "wow") {
-        cookies.test = cookie.sameSite == Ci.nsICookie.SAMESITE_LAX ? 'lax' : 'none';
+        cookies.test = sameSiteToString(cookie.sameSite);
       }
 
       if (cookie.name == "test2" && cookie.value == "wow2") {
-        cookies.test2 = cookie.sameSite == Ci.nsICookie.SAMESITE_LAX ? 'lax' : 'none';
+        cookies.test2 = sameSiteToString(cookie.sameSite);
       }
 
       if (cookie.name == "test3" && cookie.value == "wow3") {
-        cookies.test3 = cookie.sameSite == Ci.nsICookie.SAMESITE_LAX ? 'lax' : 'none';
+        cookies.test3 = sameSiteToString(cookie.sameSite);
       }
     }
 
@@ -56,7 +66,7 @@
     });
   });
 
-  is(cookies.test, "lax", "Cookie set without samesite is lax by default");
+  is(cookies.test, "unset", "Cookie set without samesite is unset");
   if (noneRequiresSecure) {
     is(cookies.test2, null, "Cookie set with samesite none, but not secure");
   } else {